src/Shortcodes/FileShortcodeProvider.php

```diff
@@ -114,11 +116,13 @@ public static function handle_shortcode($arguments, $content, $parser, $shortcod
         }
 
         // cache it for future reference
-        $cache->set($cacheKey, [
-            'markup' => $markup,
-            'filename' => $record instanceof File ? $record->getFilename() : null,
-            'hash' => $record instanceof File ? $record->getHash() : null,
-        ]);
+        if ($fileFound) {
```

Comment:
You could use `if (!$errorCode)` here perhaps, and remove the need for the extra variable?

Reply:
I find the $fileFound variables improves readability quite a bit, particularly for ImageShortcodeProvider.php